src/app/components/authorization/authorization.tsx

```diff
@@ -0,0 +1,153 @@
+import { auth } from "../../../firebase-config"
+import Button from '@mui/material/Button'
+import { Box, TextField, Typography } from '@mui/material'
+import { Link, useNavigate } from 'react-router-dom'
+import bigLogo from '../../assets/big-logo.jpg'
+import ErrorIcon from '@mui/icons-material/Error'
+
+function Authorization(): JSX.Element {
+  const [registrationEmail, setRegistrationEmail] = useState("");
+  const [registrationPassword, setRegistrationPassword] = useState("");
+
+  const [loginEmail, setLoginEmail] = useState("");
+  const [loginPassword, setLoginPassword] = useState("");
+
+  const [authorizationHendler, setAuthorizationHenadler] = useState<string>("register");
+
+  const [authorizationError, setAuthorizationError] = useState<string>("");
+
+  const registerHendlerRef = useRef<any>(null);
+  const loginHendlerRef = useRef<any>(null);
+
+  const registerEmailRef = useRef<any>(null);
+  const registerPasswordRef = useRef<any>(null);
+  const loginEmailRef = useRef<any>(null);
+  const loginPasswordRef = useRef<any>(null);
+
+  const navigate = useNavigate();
+  const register = async () => {
+    try {
+      const user = await createUserWithEmailAndPassword(
+        auth,
+        registrationEmail,
+        registrationPassword
+      );
+      const currentUser:any = auth.currentUser;
+      sendEmailVerification(currentUser);
+      auth.signOut();
+      alert("Email Send");
+      navigate("/");
+    } catch (error: any) {
+      // console.error(error.message);
+      if (error.message) {
+        setAuthorizationError("Registration Error");
+      }      
+    }
+  }
+
+  const login = async () => {   
+    try {
+      const UserCredential = await signInWithEmailAndPassword(
+        auth,
+        loginEmail,
+        loginPassword
+      );
+      navigate("/");
+    } catch (error: any) {
+      // console.error(error.message);
+      if (error.message) {
+        setAuthorizationError("Incorrect username or password");
+      }
+    }
+  }
+
+  return (
+    <Box className="moviebase-authorization-block">
+      <img src={bigLogo} alt="big-logo" className='big-logo'/>
+      <Box className='moviebase-authorization-card'>
+        <Box className="card-header-block">
+          <div className='card-header-register moviebase-authorization-card-active' 
+          onClick={() => {            
+            setAuthorizationHenadler("register");
+            setAuthorizationError("");
+            setRegistrationEmail("");
+            setRegistrationPassword("");
+            registerHendlerRef.current.classList.add("moviebase-authorization-card-active");
+            loginHendlerRef.current.classList.remove("moviebase-authorization-card-active");          
+          }} 
+          ref={registerHendlerRef}>Registration</div>
+          <div className='card-header-login' 
+          onClick={() => { 
+            setAuthorizationHenadler("login");
```

Comment:
Code duplication, separate the common logic into function. Also, do not declare a function inside of onClick prop, do it outside of JSX.